From e8e77df79b42dd2d2b93e9be5ae2baa88be70014 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Wed, 7 Aug 2024 14:07:31 +0200
Subject: [PATCH 1129/1695] plat-stm32mp2: update BSEC fuse map for STM32MP21

ECIES key is bigger on STM32MP21, update platform data accordingly.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9681f4e45cc11c4103dae44281d4cae201eb1fac plat-stm32mp2: update BSEC fuse map for STM32MP21]
Change-Id: Icc39682888adc44f914c7944fb06f1fd48178211
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/397300
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp2/platform_config.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/platform_config.h b/core/arch/arm/plat-stm32mp2/platform_config.h
index 435dd4397..54263db6e 100644
--- a/core/arch/arm/plat-stm32mp2/platform_config.h
+++ b/core/arch/arm/plat-stm32mp2/platform_config.h
@@ -107,6 +107,16 @@
 /* Console configuration */
 #define GIC_SPI_UART4			126
 
+#ifdef CFG_STM32MP21
+/*
+ * BSEC: 364 available OTPs, the other are masked
+ * - OEM FSBL keys 348 to 363 (programmable but not readable)
+ * - ECIES key: 364 to 375 (only readable by bootrom)
+ * - HWKEY: 376 to 383 (never reloadable or readable)
+ */
+#define OEM_KEY_FIRST_OTP		348
+#define STM32MP2_OTP_MAX_ID		0x16BU
+#else
 /*
  * BSEC: 368 available OTPs, the other are masked
  * - OEM FSBL keys 360 to 367 (programmable but not readable)
@@ -115,6 +125,7 @@
  */
 #define OEM_KEY_FIRST_OTP		360
 #define STM32MP2_OTP_MAX_ID		0x16FU
+#endif
 #define STM32MP2_UPPER_OTP_START	0x100U
 
 #define OTP_MAX_SIZE			(STM32MP2_OTP_MAX_ID + 1U)
-- 
2.47.2

